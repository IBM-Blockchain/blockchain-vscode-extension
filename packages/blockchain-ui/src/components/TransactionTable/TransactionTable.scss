.transaction-table-container {
    width: 100%;
}

.bx--data-table-container {
    width: 100%;
}
