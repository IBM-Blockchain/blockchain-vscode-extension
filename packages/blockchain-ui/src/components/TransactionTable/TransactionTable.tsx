import React, { Component } from 'react';
import './TransactionTable.scss';
import { Button, DataTable } from 'carbon-components-react';

const {
    TableContainer, Table,
    TableHead,
    TableHeader,
    TableRow,
    TableBody,
    TableCell,
    TableToolbar,
    TableToolbarContent,
    TableBatchActions,
    TableSelectAll,
    TableSelectRow
} = DataTable;

interface TableHeader {
    header: string;
    key: string;
}

interface TableRow {
    id: string;
    name: string;
    arguments: string;
    timestamp: string;
    result: string;
}

interface TableProps {
    id: string;
    title: string;
    description: string;
    rows: Array<TableRow> | Array<{id: string, name: string}>;
    buttonId: string;
    buttonText: string;
    buttonFunction: (viewToOpen: string) => void;
}

interface TableState {
    id: string;
    title: string;
    description: string;
    rows: Array<TableRow> | Array<{id: string, name: string}>;
    buttonId: string;
    buttonText: string;
    buttonFunction: (viewToOpen: string) => void;
}

class TransactionTable extends Component<TableProps, TableState> {

    constructor(props: Readonly<TableProps>) {
        super(props);
        this.state = {
            id: this.props.id,
            title: this.props.title,
            description: this.props.description,
            rows: (this.props.rows),
            buttonId: this.props.buttonId,
            buttonText: this.props.buttonText,
            buttonFunction: this.props.buttonFunction
        };
    }

    render(): JSX.Element {
        const tableHeaders: Array<TableHeader> = [
            { header: 'Name', key: 'name' },
            { header: 'Arguments', key: 'arguments' },
            { header: 'Timestamp', key: 'timestamp' },
            { header: 'Result', key: 'result' }
        ];
        return (
            <div className='transaction-table-container' id={this.state.id}>
                <DataTable
                    rows={this.state.rows}
                    headers={tableHeaders}
                    render={({ rows, headers, getHeaderProps, getSelectionProps, getBatchActionProps }: any): JSX.Element => (
                        <TableContainer title={this.state.title} description={this.state.description}>
                            <TableToolbar>
                                <TableBatchActions {...getBatchActionProps()}>
                                </TableBatchActions>
                                <TableToolbarContent>
                                    <Button id={this.state.buttonId} onClick={(): void => this.state.buttonFunction('create')}>{this.state.buttonText}</Button>
                                </TableToolbarContent>
                            </TableToolbar>
                            <Table>
                                <TableHead>
                                    <TableRow>
                                        <TableSelectAll {...getSelectionProps()} />
                                        {headers.map((header: TableHeader) => (
                                            <TableHeader {...getHeaderProps({ header })}>
                                                {header.header}
                                            </TableHeader>
                                        ))}
                                    </TableRow>
                                </TableHead>
                                <TableBody>
                                    {rows.map((row: any) => (
                                        <TableRow key={row.id}>
                                            <TableSelectRow {...getSelectionProps({ row })} />
                                            {row.cells.map((cell: any) => (
                                                <TableCell key={cell.id}>{cell.value}</TableCell>
                                            ))}
                                        </TableRow>
                                    ))}
                                </TableBody>
                            </Table>
                        </TableContainer>
                    )}
                />
            </div>
        );
    }
}

export default TransactionTable;
