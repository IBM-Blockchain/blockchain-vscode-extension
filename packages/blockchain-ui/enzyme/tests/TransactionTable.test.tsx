import React from 'react';
import { mount, shallow } from 'enzyme';
import TransactionTable from '../../src/components/TransactionTable/TransactionTable';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
chai.should();
chai.use(sinonChai);

describe('TransactionTable component', () => {

    let mySandBox: sinon.SinonSandbox;
    let mockTableProps: any;

    beforeEach(async () => {
        mySandBox = sinon.createSandbox();

        mockTableProps = {
            title: 'Mock Table',
            description: 'A table that has been mocked.',
            rows: {
                id: '1',
                name: 'createAsset',
                arguments: 'AssetID: myAsset01',
                timestamp: '1:15pm',
                result: 'âœ…'
            },
            buttonText: 'Click me',
            buttonFunction: mySandBox.stub()
        };
    });

    afterEach(async () => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', async () => {
        const component: any = shallow(<TransactionTable {...mockTableProps}/>);
        expect(component.getElements()).toMatchSnapshot();
    });
});
