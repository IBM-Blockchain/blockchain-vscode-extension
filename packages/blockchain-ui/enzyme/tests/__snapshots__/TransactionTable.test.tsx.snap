// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionTable component should render the expected snapshot 1`] = `
Array [
  <div
    className="transaction-table-container"
  >
    <DataTable
      filterRows={[Function]}
      headers={
        Array [
          Object {
            "header": "Name",
            "key": "name",
          },
          Object {
            "header": "Arguments",
            "key": "arguments",
          },
          Object {
            "header": "Timestamp",
            "key": "timestamp",
          },
          Object {
            "header": "Result",
            "key": "result",
          },
        ]
      }
      locale="en"
      render={[Function]}
      rows={
        Object {
          "arguments": "AssetID: myAsset01",
          "id": "1",
          "name": "createAsset",
          "result": "âœ…",
          "timestamp": "1:15pm",
        }
      }
      sortRow={[Function]}
      translateWithId={[Function]}
    />
  </div>,
]
`;
